### PR TITLE
Test images to docs

### PR DESCRIPTION
**Pull request by typemytype**
this is a big one...

changing a lot of docs code samples to produce beter image, as they are directly used in the website

### CODE REVIEW

**Comment by typemytype** [comment]
and after this one Im going to take a small drawBot break and make a new release for public, a lot has happend

**Comment by typemytype** [comment]
this is related with https://github.com/typemytype/drawbot/issues/151

**Comment by justvanrossum** [comment]
Kijk wat er in Py3 met retina uitkomt:



Moet dit zijn:


Ik ga kijken of ik kan vinden waar dat gebeurt. Daar moet wellicht ook ergens een _makeBitmapImageRep() gedaan worden.

Just

**Comment by justvanrossum** [comment]
De volgende fix maakt de output al veel beter kloppend:

diff --git a/drawBot/context/tools/imageObject.py b/drawBot/context/tools/imageObject.py
index fdfe054..353a799 100644
--- a/drawBot/context/tools/imageObject.py
+++ b/drawBot/context/tools/imageObject.py
@@ -6,6 +6,7 @@ import os
 
 from drawBot.misc import DrawBotError, optimizePath
 from fontTools.misc.py23 import basestring
+from drawBot.context.imageContext import _makeBitmapImageRep
 
 
 class ImageObject(object):
@@ -121,7 +122,7 @@ class ImageObject(object):
         # create an image
         im = AppKit.NSImage.alloc().initWithData_(page.dataRepresentation())
         # create an CIImage object
-        ciImage = AppKit.CIImage.imageWithData_(im.TIFFRepresentation())
+        ciImage = AppKit.CIImage.imageWithData_(_makeBitmapImageRep(im).TIFFRepresentation())
         # merge it with the already set data, if there already an image
         self._merge(ciImage)
 

Maar de uitvoer is toch subtiel anders met Py3+retina, diff plaatje (zonder alpha):



Ik denk omdat er intern toch ergens met de retina reolutie gerekend wordt.

Heb jij nog een idee waar te kijken? Liefst wil ik fuzzy-image-vergelijken nog even uitstellen :)

Just

**Comment by typemytype** [comment]
will test tomorrow on a non retina screen and merge from there

**Comment by justvanrossum** [comment]
This unfortunately still doesn't fix the test failure on Retina with Py3 in `test_imageObject()`. Do you want me to go ahead and add a slightly tolerant image compare function? I have it working already, just need to integrate. Will then require PIL/Pillow to be able to run the tests.

(Ha, this would be a case for a PR on a PR :) )

**Comment by typemytype** [comment]
I only get a different test_imageObject in py3...

**Comment by justvanrossum** [comment]
> I only get a different test_imageObject in py3...

That's what I said...

**Comment by justvanrossum** [comment]
I just added fuzzy image comparison; we could enable that only in specific cases, similar to `dontSaveImage`.

**Comment by typemytype** [comment]
yeah, enable `dontSaveImage` only in specific cases that we know of and that we grant permission to use the fuzzy comparing instead of hard data comparing

**Comment by typemytype** [comment]
super!!! I guess this PR is good to go :)

File: tests/testExamples.py [hunk 1]
```diff
@@ -140,20 +160,31 @@ def mockSaveImage(path, **options):
...
+                self.assertImagesSimilar(imagePath, expectedImagePath)
+            else:
+                self.assertFilesEqual(imagePath, expectedImagePath)
 
     return test
 
 
 skip = {}
 expectedFailures = {}
 dontSaveImage = ["test_imageSize"]
```
Comment (typemytype) [inline tests/testExamples.py]: nitpicking: is there a reason why this is a list and not a set?

**Comment by justvanrossum** [comment]
Yep, go ahead and merge!

**Pull request merged**